version: '3.7'

services:
  app:
    image: chaosdorf/mete:latest
    environment:
      - RAILS_ENV=production
      - SECRET_KEY_BASE=yoloyolo123
      - DATABASE_URL=postgresql://mete:mete@db:5432/mete?pool=5&timeout=5000
    secrets:
      - source: METE_SENTRY_DSN
        target: SENTRY_DSN
    volumes:
      - app-staticfiles:/app/public
    networks:
      - internal
      - traefik
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: on-failure
      labels:
        - traefik.docker.network=traefik_net
        - traefik.port=80
        - traefik.frontend.rule=Host:mete.chaosdorf.space,mete
  db:
    image: postgres:10-alpine
    environment:
      POSTGRES_USER: mete
      POSTGRES_PASSWORD: mete
      POSTGRES_DB: nonexistant # mete relies on the actual db not yet existing on first start
    volumes:
      - db-data:/var/lib/postgresql/data
    networks:
      - internal
    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: on-failure

volumes:
  app-staticfiles:
  db-data:

networks:
  internal:
    driver: overlay
  traefik:
    driver: overlay
    external: true
    name: traefik_net

secrets:
  METE_SENTRY_DSN:
    external: true
